dev/core/src/com/google/gwt/dev/jjs/ast/JFieldRef.java

```diff
@@ -92,6 +92,9 @@ public boolean hasSideEffects() {
     if (hasClinit()) {
       return true;
     }
+    if (getField().isJsNative()) {
+      return false;
+    }
```

Comment:
This is just plain backwards - I think I inverted it at some point for testing purposes - but I think I'd prefer to remove it entirely.

The idea was supposed to be as a way to trigger ImplementJsVarargs own "if this has side effects I should only eval it once" in a nicer way than assuming it recognizes all qualified name references (since its failure to do so caused the original bug). However, this ends up having other side effects, like preventing a lot of dead code elimination from working correctly.

The "hasSideEffects" check itself in ImplementJsVarargs isn't wrong, but this particular attempt at a safeguard is. The comment in ImplementJsVarargs should also be corrected.